### DISCUSSION TITLE
Unable to upgrade renovate-runner due to `Ignoring upgrade collision`.

### DISCUSSION THREAD

**Discussion by himanshumehta1114** [renovatebot/renovate#10029] [Request Help]
Hello Folks, 

I am not receiving updates from the last few days,  it returns `Ignoring Upgrade Collision` message with the latest version as `previousNewValue`. Can you confirm what is meant by `previousNewValue`, `thisNewValue`, and `currentValue` ? It's not very understandable by code [here](https://github.com/renovatebot/renovate/blob/b8e36daa87e4275caa5b66bffa6fe3f4d5d8a77e/lib/workers/repository/updates/branchify.ts)

If it's detecting the newer version, shouldn't it delete the older branch and create a new one with the latest update?

I am using the following configuration:
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "github>renovatebot/.github"
  ],
  "assignees": [],
  "gitLabAutomerge": true,
  "packageRules": [
    {
      "description": "Automerge renovate minor and patch updates",
      "matchPackageNames": [
        "renovate/renovate"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "automerge": true,
      "branchTopic": "{{{depNameSanitized}}}-{{{currentValue}}}"
    },
    {
      "description": "Allow updates after 3 days (exclude renovate)",
      "excludePackageNames": [
        "renovate/renovate"
      ],
      "separateMinorPatch": true,
      "stabilityDays": 3
    }
  ],
  "gitlabci": {
    "fileMatch": [
      "\\.gitlab-ci\\.yml$"
    ]
  }
}

```

## Error message log 👉
![image](https://user-images.githubusercontent.com/25463496/118602018-bce26900-b7cf-11eb-869b-38cd346dfe1f.png)

## Dependency dashboard 👉
![image](https://user-images.githubusercontent.com/25463496/118602062-cff53900-b7cf-11eb-8642-98695e8ba021.png)

## Branch not being updated 👉
![image](https://user-images.githubusercontent.com/25463496/118602087-daafce00-b7cf-11eb-833b-c1ce9b9d98e3.png)

**Comment by rarkins** [comment]
IIRC this happens when there is an attempt to upgrade the same dependency in the same file to two different values. Possibly your overriding of `branchTopic` is causing it - why did you change it?

**Reply by himanshumehta1114** [reply]
Well, I did not change the `branchTopic`. It's the same since I setup the `renovate-runner`. But I changed the RENOVATE_TOKEN, can that be an issue?

**Reply by viceice** [reply]
if it's a new user, then the old branch author is blocking renovate to reuse the branch use `gitIgnoredAuthors` to ignore your old renovate git author

https://docs.renovatebot.com/configuration-options/#gitignoredauthors

**Reply by himanshumehta1114** [reply]
> IIRC this happens when there is an attempt to upgrade the same dependency in the same file to two different values. Possibly your overriding of `branchTopic` is causing it - why did you change it?

Thanks for pointing it out. I forked the project from [here](https://gitlab.com/renovate-bot/renovate-runner). It had the branchTopic configure. Removing it fixes the issue. May I know how `branchTopic` affects the updates, does Renovate picks the version from `branchTopic` ?

**Reply by rarkins** [reply]
`branchTopic` is part of `branchName`. Don't change either of them unless you  know what you're doing

**Reply by himanshumehta1114** [reply]
Ah okay, sure.

One last thing, according to following configuration `renovate/renovate` minor & patch updates should be automerged. All except `renovate/renovate` should go into pending state for 3 days right?

But still my `renovate/renovate` docker tag updates go into pending state.

![image](https://user-images.githubusercontent.com/25463496/118611047-0cc62d80-b7da-11eb-8d39-814a67f5218a.png)


```
"packageRules": [
    {
      "description": "Automerge renovate minor and patch updates",
      "matchPackageNames": [
        "renovate/renovate"
      ],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "automerge": true
    },
    {
      "description": "Allow updates after 3 days (exclude renovate)",
      "excludePackageNames": [
        "renovate/renovate"
      ],
      "separateMinorPatch": true,
      "stabilityDays": 3
    }
  ],
```

**Reply by viceice** [reply]
you need a ci pipeline, which runs on renovate branches, otherwise renovate would never automerge. We are using a dry-run against some test repos to make sure a new renovate version won't break an older.

⚠️ You can disable status check, but it's not recommend and can break your default / target branch

https://docs.renovatebot.com/configuration-options/#requiredstatuschecks

**Reply by himanshumehta1114** [reply]
I have a similar pipeline setup which dry-run on all branches except master to verify & merge PR to master. That makes sense. 

But this configuration says `Allow updates after 3 days (exclude renovate)` which means packages except `renovate/renovate` should go in pending state. But according to dependency dashboard screenshot(shared above),  the `renovate/renovate` docker tag is currently in pending which means it respects stabilityDays though it is`renovate/renovate` package. 

Which is not intended right ? Please correct me if I'm wrong.

**Reply by rarkins** [reply]
If the version being updated to is less than three days old then you'll see a pending status on the branch. We tend to release Renovate more frequently than 3 days though, so you may see permanent pending status. You can experiment with setting https://docs.renovatebot.com/configuration-options/#internalchecksfilter to "strict"

**Reply by himanshumehta1114** [reply]
But it should exclude branches with `renovate/renovate` package updates. According to following configuration:

```
{
      "description": "Allow updates after 3 days (exclude renovate)",
      "excludePackageNames": [
        "renovate/renovate"
      ],
      "separateMinorPatch": true,
      "stabilityDays": 3
    }
```

**Reply by rarkins** [reply]
See https://github.com/renovatebot/renovate/discussions/10029#discussioncomment-751524

**Reply by viceice** [reply]
Please provide debug logs for futher help

**Reply by himanshumehta1114** [reply]
Thanks for explaining it. I understood the point that CI will auto-merge after the stability period is over for the respective release.

But what I'm saying is `v25.21.13` was released 6 hours ago and the [renovate-runner](https://gitlab.com/renovate-bot/renovate-runner/-/issues/2) dependency dashboard says the dependency status check is pending. But I see it has already merged to master. Though the stability period is not over (i.e. 1 day) in this case - I'm referring to this configuration file - [renovate.json](https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/renovate.json) 

Does this mean the `stabilityDays` configuration is ignored here? Please correct me if I'm wrong.

**Reply by viceice** [reply]
The dependency dashboard is outdated after automerge, because renovate stops early atfer automerge. So renovate needs another run to update the dashboard.

**Reply by viceice** [reply]
Also we use a special configuration for renovate docker image.

https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/renovate.json#L52
```json
{
      "description": "Allow renovate updates after 1 day",
      "matchPackageNames": [
        "renovate/renovate"
      ],
      "stabilityDays": 1
    },
```

with [internalChecksFilter=strict](https://docs.renovatebot.com/configuration-options/#internalchecksfilter).
https://github.com/renovatebot/.github/blob/930726ea9db00b581216bc7618a832dfcd81155a/default.json#L19

so renovate updates are delayed by one day.

**Reply by himanshumehta1114** [reply]
Yes, but here it's not delayed 😅 . The release was made 6 hours ago and right now it's merged to [master](https://gitlab.com/renovate-bot/renovate-runner/-/commits/master). The gap between release and merging here is 5 hours which is less than 1 day.

**Reply by viceice** [reply]
I think it's happening because the docker image got a digest update, so new release date 😕 Also when the pr is already open, it will ignore `stabilityDays` with `internalChecksFilter`. 🤔
